srtcore/logging.h

```diff
@@ -133,21 +133,36 @@ struct SRT_API LogDispatcher
 private:
     int fa;
     LogLevel::type level;
-    std::string prefix;
+    static const size_t MAX_PREFIX_SIZE = 32;
+    char prefix[MAX_PREFIX_SIZE+1];
     LogConfig* src_config;
     pthread_mutex_t mutex;
 
     bool isset(int flg) { return (src_config->flags & flg) != 0; }
 
 public:
 
-    LogDispatcher(int functional_area, LogLevel::type log_level, const std::string& pfx, LogConfig& config):
+    LogDispatcher(int functional_area, LogLevel::type log_level, const char* your_pfx,
+            const char* logger_pfx /*[[nullable]]*/, LogConfig& config):
         fa(functional_area),
         level(log_level),
-        prefix(pfx),
-        //enabled(false),
         src_config(&config)
     {
+        // XXX stpcpy desired, but not enough portable
+        // Composing the exact prefix is not critical, so simply
+        // cut the prefix, if the lenght is exceeded
+
+        // See Logger::Logger; we know this has normally 2 characters,
+        // except !!FATAL!!, which has 9. Still less than 32.
+        strcpy(prefix, your_pfx);
```

Comment:
would it be worth adding a debug assert here to check that your_pfx is less than 32?

Reply:
A static assert might make some sense (it's C++11 though), but I'd see a debug assert exaggerated here. Note that this constructor is used only at the `Logger` constructor at line 378. It's even in the same file and I doubt anyone will do any messup here in future.

Reply:
ok, makes sense